doc/manual/src/architecture/store/reference-scanning.md

```diff
@@ -0,0 +1,57 @@
+# Reference scanning
+
+As discussed in a previous section, a store object is made up of file system objects and references to other store objects.
+
+Suppose we are trying to create a new store object that *ought* to have references, but we only have plain file system data.
+For example, imagine the file system data is a "shared library" (so, dll, dylib) that depends on other shared libraries, and those other shared libraries are already inside other store objects.
+How might we do this?
+
+One way would be to have the user manually write down this information.
+Another way would be to teach Nix to parse those files' headers and convert the "domain specific" dependency information within them.
+Both of these are a huge amount of work, both initially and as ongoing maintenance, and therefore not very attractive options.
+
+Nix instead opts to merely scan the file for store paths without making any attempt to parse it.
+This "simple, stupid" approach actually works quite well most of the time, and imposes very little burden on the end user.
+
+## Why it works
+
+Store objects are, as previously mentioned, mounted at `<store-dir>/<hash>-<name>` when they are exposed to the file system.
+This is the textual form of a [store path](./paths.md).
+When this happens depends on the type of store, but it *must* happen during building, as regular Unix processes are being run.
```

Comment:
When what happens? What is "this" here? Reference scanning or mounting?
Also, who runs regular Unix processes? Are there irregular Unix processes?

Reply:
```suggestion
When this mounting happens depends on the type of store, but it *must* happen prior to building.
Regular Unix processes that don't necessarily know about Nix are being run, and they need to access the store object inputs of the build.
By mounting the closure of all input store objects, we make them accessible to regular processes as plain directories.
```